File: assignments/week-8/0435-non-overlapping-intervals/solution.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        # Logic: Sort the intervals based on end_time
        # and then remove the overlapping intervals

        intervals.sort(key=lambda x:x[1])

        min_intervals_count = 0
        prev_interval = intervals[0]

        for idx in range(1, len(intervals)):
            interval = intervals[idx]
            # Is interval overlapping?
            if prev_interval[1] > interval[0]:
                min_intervals_count += 1
            else:
                prev_interval = interval
    
        return min_intervals_count

        

File: assignments/week-8/0435-non-overlapping-intervals/solution.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[1], b[1]));
        int n = intervals.length;

        int minIntervalsCount = 0;
        int[] prevInterval = intervals[0];

        for(int i = 1; i < n; i++){
            int[] currInterval = intervals[i];

            if (prevInterval[1] > currInterval[0]){
                minIntervalsCount += 1;
            }
            else{
                prevInterval = currInterval;
            }
        }

        return minIntervalsCount;
    }
}